WordPress/openverse
# Add Django UI admin notice for next staging database restore

## Fixes



Fixes #1990 by @AetherUnbound

## Description



This PR overrides the Admin UI `base_site.html` template ([as described here](https://docs.djangoproject.com/en/4.2/howto/overriding-templates/#extending-an-overridden-template)) and leverages the messages list to always display a message about the number of days until the next staging database restore (along with a link for how to prevent the restore).

I used some really rudimentary Javascript to calculate the days-until value for this, so if there are any better or easier ways to do that please feel free to drop them!

Unfortunately the Django Admin UI CSS doesn't provide any distinction between success/debug/info messages (all show up as a green check) so `warning` seemed most appropriate here.

![image](https://github.com/WordPress/openverse/assets/10214785/443091da-4197-4a5d-96c8-dbfdaecf7891)


## Testing Instructions


1. `just a`
1. Visit http://localhost:50280/admin/ and observe the warning
1. Visit http://localhost:50280/admin/api/imagereport/ and observe that the warning persists
1. Modify an image report value and observe that the save message exists alongside the staging DB warning
1. Change or set the `ENVIRONMENT` value in `api/.env` to `production` and run `just a`
1. Perform the above actions and observe that the warning is no longer present.


## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
